python/pyspark/tests.py

```diff
@@ -644,6 +644,18 @@ def test_cartesian_chaining(self):
             set([(x, (y, y)) for x in range(10) for y in range(10)])
         )
 
+    def test_zip_chaining(self):
+        # Tests for SPARK-21985
+        rdd = self.sc.parallelize('abc')
```

Comment:
I'd set the explicit number of partitions because `zip` reserializes it depending on this.